src/viking.js

```diff
@@ -1,11 +1,107 @@
 // Soldier
-class Soldier {}
-
-// Viking
-class Viking {}
-
-// Saxon
-class Saxon {}
-
-// War
-class War {}
+class Soldier {
+    constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+    }
+    
+    attack() {
+    return this.strength;
+    }
+    
+    receiveDamage(damage) {
+    this.health -= damage;
+    }
+    }
+    
+    // Viking
+    class Viking extends Soldier {
+    constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+    }
+    
+    receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+    return `${this.name} has received ${damage} points of damage`;
+    } else {
+    return `${this.name} has died in act of combat`;
+    }
+    }
+    
+    battleCry() {
+    return "Odin Owns You All!";
+    }
+    }
+    
+    // Saxon
+    class Saxon extends Soldier {
+    constructor(health, strength) {
+    super(health, strength);
```

Comment:
Since we are not passing a new parameter for the class Saxon we don't need the constructor neither the super:

```
class Saxon extends Soldier {
  receiveDamage(damage) {
    this.health -= damage;
    if (this.health > 0) {
      return `A Saxon has received ${damage} points of damage`;
    } else {
      return `A Saxon has died in combat`;
    }
  }
}
```